test-integration/helpers/student-dashboard.coffee

```diff
@@ -0,0 +1,42 @@
+selenium = require 'selenium-webdriver'
+{TestHelper} = require './test-element'
+_ = require 'underscore'
+
+
+class Event extends TestHelper
+
+  elementRefs:
+    title:
+      css: '.title'
+    progress:
+      css: '.feedback'
+
+  constructor: (test, eventId) ->
+    super(test, ".task[data-event-id='#{eventId}']")
+
+  @fromElement: (test, element) ->
+    console.log "from el"
+    element.getAttribute('data-event-id').then (el) ->
+      new Event(test, el)
+
+class Dashboard extends TestHelper
+
+  elementRefs:
+    progressGuide:
+      css: '.progress-guide'
+    thisWeek:
+      css: '.panel.-this-week'
+    visibleEvents:
+      css: '.tab-pane.active .task'
+
+  constructor: (test) ->
+    super(test, '.student-dashboard')
+
+  getVisibleEventHelpers: ->
+    makeEvent = _.partial(Event.fromElement, @test)
+    @el.visibleEvents.getAll().then( (tasks) ->
+      Promise.all( _.map(tasks, makeEvent ) )
```

Comment:
Yep, we should do that.  In this case it doesn't really matter because `Promise.all` just resolves all the existing (selenium) promises.  It's still a good idea to use the selenium though.  I'll switch it.